### ISSUE TITLE
Support escape space by backslash in command line

### ISSUE THREAD

**Issue by PRNDA** [nushell/nushell#12464]
### Related problem

I use Mac as my primary system, and Mac's Terminal.app is perfectly integrated with the whole system, for example, if I drag (or copy-and-paste) files/folders into a Terminal window, it will paste the full file path into the command line, and the space characters in the file path is escaped by a backslash automatically! 

Like this: `diff /Users/mac/Downloads/a\ hourly\ list\ 1.txt  /Users/mac/another \dir/and\ another\ file.txt`

This is a very convenient feature for a man who uses the command line heavily, especially when you need to copy-and-paste **_a lot of_** file paths.

But `nushell` does not support escape space by backslash( \<space>), this makes my daily work a bit harder, because I use copy-and-paste files/folders into the Terminal now and then all the day.

![Image](https://github.com/user-attachments/assets/17e1b29c-f4c0-449f-8ec1-59c147e46046)

### Describe the solution you'd like

I would like `nushell` to support escape space by backslash like this: `/Users/mac/Downloads/a\ hourly\ list.txt`

**Comment by devyn** [comment]
While your use case makes sense, I wonder if it might conflict with our support for Windows paths.

**Comment by KAAtheWiseGit** [comment]
This would also require changing the semantics of unquoted strings, which do not support escapes.  So, unlike in other shells, `print \t` will output `\t` literally.

**Comment by bratorange** [comment]
> We are commited that Nushell does not display platform differences for the core language, so configuration options, switches, or platform detection are out of the question for syntax level questions.

But would a config option to change this behavior compatible with the design concept / community?

**Comment by oderwat** [comment]
I would also like to have this as an option. Especially becase dropping a file into a Ghostty terminal seems to always use the escapes. It is not just space that gets escaped there are also escaped parentesis and other characters I suppose. I understand the Window problem and also that one does not want to have different behavior. But then I wonder if this is really something that needs to be implemented on the "core" parser. Why not having it as an option for the terminal input only? I know this is very special, but if you are in the position that this is how it works, you get really tired of having to find ways to insert a path which is so easy to insert into (all?) other shells by just dropping it into the terminal pane. But I have to say that I was looking if one can change Ghostty before I ended up here.

**Comment by PRNDA** [comment]
> also escaped parenthesis and other characters

Yes, all special characters are automatically escaped.

A MUST-HAVE feature.

**Comment by KAAtheWiseGit** [comment]
If these terminals use bracketed paste mode, then it should be possible to catch these events and escape the whole path using Nushell-native quotes, much like the auto-completion does

**Comment by jobukkit** [comment]
This issue affects the Linux version of Ghostty too, so it's not MacOS-specific.

For iTerm2, you can fix it by enabling "Escape file names with quotes instead of backslashes" under the advanced settings.

This is the same issue as https://github.com/nushell/nushell/issues/11944.

**Comment by ronisbr** [comment]
I am not sure if I am doing something wrong, but even after that PR in v0.113.0, this bug still exists:

<img width="523" height="360" alt="Image" src="https://github.com/user-attachments/assets/051137c7-cfc7-48c1-8c7a-33d960b74f10" />

**Comment by fdncred** [comment]
> I am not sure if I am doing something wrong, but even after that PR in v0.113.0, this bug still exists:

Why would that work when the PR specifically says it requires double quotes?

**Comment by ronisbr** [comment]
> Why would that work when the PR specifically says it requires double quotes?

Same problem:

<img width="940" height="737" alt="Image" src="https://github.com/user-attachments/assets/c93efb1f-9ab1-4bd6-9046-712c58831e3c" />

**Comment by fdncred** [comment]
Right, I believe that is working as expected. Paths with spaces require backticks. I don't believe spaces are in the list of escape-able characters.
https://github.com/nushell/nushell/blob/e7600202b54bfaf26f66a111f400921df7e8ab81/crates/nu-parser/src/parser.rs#L3502-L3519

**Comment by ronisbr** [comment]
So this issue should continue open... escaping spaces is the problem that OP reported and I am facing too.

**Comment by fdncred** [comment]
"Special characters" are escaped. I agree the title doesn't match the OP's description.

But we could add space if we wanted. I'm just not sure we want this. A few problems I noticed with this is we'd probably also have to change completions, make `~` and ndots work. Plus the OP wanted it to work without double quotes. I don't really see this happening. 

<img width="3019" height="961" alt="Image" src="https://github.com/user-attachments/assets/91a6636f-53ab-42e3-a704-41d9d1077b71" />

**Comment by ronisbr** [comment]
I am not sure about OP, but my use case would be solved if the escape for spaces works inside double-quotes only.

**Comment by oderwat** [comment]
It needs to work without double quotes to be useful when dragging files into a terminal on OSX. At least with original `Terminal`  and with `Ghostty`. That said, AFAIK "\ " is a not a valid escape sequence in double quotes. But you may want to add a rule "`f it is unknown, just ignore the \`. I think that is done in some implementations.

Example for a drag&drop file path in Ghostty:

<img width="347" height="46" alt="Image" src="https://github.com/user-attachments/assets/248ddb6c-ef0a-4a29-bacc-15fab0fd8369" />

**Comment by fdncred** [comment]
The problem isn't really the escaping. That's easy to add. The problem is with everything that uses paths with spaces like completions, expanding ~, expanding ndots, command arguments, custom command arguments, highlighting, etc. It's kind of a can of worms.

**Comment by jobukkit** [comment]
> It needs to work without double quotes to be useful when dragging files into a terminal on OSX.

Being able to type a ", drag and drop, then type another " would arguably be enough already. Some Linux terminals like Terminology make you do this for all shells.

**Comment by oderwat** [comment]
> ... "dragging files" ...

> Being able to type a ", drag and drop, then type another " would arguably be enough already.

This would work for a single file being dragged. But you can drag multiple ones. Which makes it even more useful.

**Comment by fdncred** [comment]
How would this work in Windows?

**Comment by PRNDA** [comment]
> > It needs to work without double quotes to be useful when dragging files into a terminal on OSX.
> 
> Being able to type a ", drag and drop, then type another " would arguably be enough already. Some Linux terminals like Terminology make you do this for all shells.

If you just want this, than you don't need to do anything at all, because you can already do this:

`ls "spaces in file names.txt"`

**Comment by oderwat** [comment]
> I you just want this, than you don't need to do anything at all, because you can already do this:
> 
> `ls "spaces in file names.txt"`

I think what he means is that you can type the two " and insert the version that has "\ " in it. Result would be:

"spaces\ in\ file\ names.txt"

And the "\ " is supported as space escape for a string. Basically striping the \ in front of a space.

And it would indeed be "better" because right now one needs to go and remove those \ manually to not type everything again. But this is obviously a pain. I actually started to the "copy as path" in the bottom bar when I need a file path in the terminal. I would love to go back to just drag&drop.

**Comment by ronisbr** [comment]
Indeed, allowing spaces to be escaped inside quotes will help a lot. In this case, we can add quotes and drag files and it will work. Right now, it is impossible.

**Comment by ronisbr** [comment]
Hi!

Any conclusions about this issue? I can make a PR if devs decide what direction we should take.

**Comment by fdncred** [comment]
have at it but it can't be a breaking change and it has to work on Windows, MacOS, and Linux like the rest of nushell.

**Comment by ronisbr** [comment]
@fdncred Done! The idea is that now we can type `"` drag-and-drop and `"`. It is not breaking because it was an error before.
